### ISSUE TITLE
Compactor doesn't work as expected

### ISSUE THREAD

**Issue by LinTechSo** [grafana/loki#6953]
## Describe the bug
 Trying to delete the log by api request on compactor , but looks like there is no such a api.
``curl -g -X POST 'http://localhost:3100/loki/api/v1/delete?query={cluster="ks3"}&start=1660817486&end=1661218287' -H 'x-orgid-scope: test'``
404 page not found
``curl -g -X POST 'http://localhost:3100/loki/api/admin/delete?match[]={cluster="ks3"}&start=1660817486&end=1661218287' -H 'x-scope-orgid: test'``
404 page not found

and also my retention time configuration doesn't work
`level=error ts=2022-08-23T03:20:51.434651347Z caller=retention.go:264 msg="error deleting chunk" chunkID=test/5852906667dcf873:182c277e095:182c2e632c5:581ed5c8 err="RequestError: send request failed\ncaused by: Delete \"http://loki-distributed-minio.test.svc:9000/loki/test/5852906667dcf873%3A182c277e095%3A182c2e632c5%3A581ed5c8\": dial tcp 10.97.241.250:9000: i/o timeout"
level=error ts=2022-08-23T03:21:21.43764404Z caller=retention.go:264 msg="error deleting chunk" chunkID=test/5852906667dcf873:182c2e636ac:182c354810d:cb819905 err="RequestError: send request failed\ncaused by: Delete \"http://loki-distributed-minio.test.svc:9000/loki/test/5852906667dcf873%3A182c2e636ac%3A182c354810d%3Acb819905\": dial tcp 10.97.241.250:9000: i/o timeout"
level=error ts=2022-08-23T03:21:21.437645847Z caller=retention.go:264 msg="error deleting chunk" chunkID=test/5852906667dcf873:182c277e095:182c2e632c5:581ed5c8 err="RequestError: send request failed\ncaused by: Delete \"http://loki-distributed-minio.test.svc:9000/loki/test/5852906667dcf873%3A182c277e095%3A182c2e632c5%3A581ed5c8\": dial tcp 10.97.241.250:9000: i/o timeout"`

My Loki version is 2.6.1 and i am using distributed helm chart

my configuration:

```
    auth_enabled: true
    chunk_store_config:
      max_look_back_period: 0s
    compactor:
      compaction_interval: 5m
      retention_delete_delay: 1h
      retention_delete_worker_count: 1200
      retention_enabled: true
      shared_store: s3
      working_directory: /var/loki/compactor
    distributor:
      ring:
        kvstore:
          store: memberlist
    frontend:
      compress_responses: true
      log_queries_longer_than: 5s
      max_outstanding_per_tenant: 10000
      tail_proxy_url: http://querier:3100
    frontend_worker:
      frontend_address: frontend:9095
      grpc_client_config:
        max_recv_msg_size: 1084857600
        max_send_msg_size: 1084857600
    ingester:
      chunk_encoding: snappy
      chunk_idle_period: 1h
      chunk_target_size: 2621440
      lifecycler:
        ring:
          kvstore:
            store: memberlist
          replication_factor: 1
      max_transfer_retries: 0
      wal:
        dir: /var/loki/wal
    ingester_client:
      grpc_client_config:
        max_recv_msg_size: 1084857600
        max_send_msg_size: 1084857600
    limits_config:
      cardinality_limit: 300000
      enforce_metric_name: false
      ingestion_burst_size_mb: 10
      ingestion_rate_mb: 8
      max_cache_freshness_per_query: 10m
      max_query_parallelism: 96
      per_stream_rate_limit: 80MB
      per_stream_rate_limit_burst: 100MB
      reject_old_samples: true
      reject_old_samples_max_age: 168h
      retention_period: 24h
      split_queries_by_interval: 15m
    memberlist:
      join_members:
      - memberlist
    query_scheduler:
      grpc_client_config:
        max_recv_msg_size: 1084857600
        max_send_msg_size: 1084857600
      max_outstanding_requests_per_tenant: 10000
    runtime_config:
      file: /opt/overrides.yaml
      period: 20s
    schema_config:
      configs:
      - from: "2020-09-07"
        index:
          period: 24h
          prefix: loki_index_
        object_store: s3
        schema: v11
        store: boltdb-shipper
    server:
      grpc_server_max_recv_msg_size: 1084957600
      grpc_server_max_send_msg_size: 1084957600
      http_listen_port: 3100
      http_server_read_timeout: 600s
      http_server_write_timeout: 600s
    storage_config:
      aws:
        access_key_id: xxxxxxxxxxx
        bucketnames: loki
        endpoint: loki-distributed-minio.test.svc:9000
        http_config:
          response_header_timeout: 600s
        insecure: true
        s3forcepathstyle: true
        secret_access_key: yyyyyyyy
      boltdb_shipper:
        active_index_directory: /var/loki/index
        cache_location: /var/loki/cache
        cache_ttl: 12h
        index_gateway_client:
          server_address: dns:///index-gateway:9095
        shared_store: s3
    table_manager:
      retention_deletes_enabled: false
      retention_period: 0s
```

## Expected behavior
both log deletion api and retention time work on my logs

**Comment by rt005744** [comment]
I have the same problem

**Comment by rt005744** [comment]
I have encountered the same problem. If you solve this problem, please tell me the solution. Thank you very much

**Comment by LinTechSo** [comment]
Hi @rt005744 .
The problem still persists which is a pity.
I rollbacked to Loki version 2.5.0 and  log deletetion request api works.

**Comment by LinTechSo** [comment]
i added `allow_deletes` to `limits_config` but the problem still persists.

**Comment by MichelHollands** [comment]
@LinTechSo Your configuration seems to be missing the `deletion_mode` setting. Can you try again with that one set to `filter-and-delete`?

**Comment by LinTechSo** [comment]
> @LinTechSo Your configuration seems to be missing the `deletion_mode` setting. Can you try again with that one set to `filter-and-delete`?

and this error coming 
```
level=info ts=2022-08-24T08:03:36.621499751Z caller=main.go:103 msg="Starting Loki" version="(version=2.6.1, branch=HEAD, revision=6bd05c9a4)"
level=info ts=2022-08-24T08:03:36.621995635Z caller=server.go:288 http=[::]:3100 grpc=[::]:9095 msg="server listening on addresses"
level=warn ts=2022-08-24T08:03:36.623574914Z caller=experimental.go:20 msg="experimental feature in use" feature="In-memory (FIFO) cache - chunksfifocache"
query filtering for deletes requires 'compactor_address' to be configured
error initialising module: querier
github.com/grafana/dskit/modules.(*Manager).initModule
	/src/loki/vendor/github.com/grafana/dskit/modules/modules.go:122
github.com/grafana/dskit/modules.(*Manager).InitModuleServices
	/src/loki/vendor/github.com/grafana/dskit/modules/modules.go:92
github.com/grafana/loki/pkg/loki.(*Loki).Run
	/src/loki/pkg/loki/loki.go:341
main.main
	/src/loki/cmd/loki/main.go:105
runtime.main
	/usr/local/go/src/runtime/proc.go:255
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1581
level=error ts=2022-08-24T08:03:36.629057313Z caller=log.go:103 msg="error running loki" err="query filtering for deletes requires 'compactor_address' to be configured\nerror initialising module: querier\ngithub.com/grafana/dskit/modules.(*Manager).initModule\n\t/src/loki/vendor/github.com/grafana/dskit/modules/modules.go:122\ngithub.com/grafana/dskit/modules.(*Manager).InitModuleServices\n\t/src/loki/vendor/github.com/grafana/dskit/modules/modules.go:92\ngithub.com/grafana/loki/pkg/loki.(*Loki).Run\n\t/src/loki/pkg/loki/loki.go:341\nmain.main\n\t/src/loki/cmd/loki/main.go:105\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:255\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581"
```

**Comment by LinTechSo** [comment]
and my question is if I set `deletion_mode` to `filter-and-delete`, does it affect the total retention configuration? because I want Loki to delete whole logs after 24h

**Comment by MichelHollands** [comment]
Yes, `compactor_address` should be set to the compactor.

Retention and deletion are 2 separate things. You don't need deletion enabled for compaction to work. The retention config docs are here: https://grafana.com/docs/loki/latest/operations/storage/retention/#retention-configuration.

**Comment by LinTechSo** [comment]
@MichelHollands Thank you so much for your help. my last question is do you have any idea about my second problem with retention? `msg="error deleting chunk"`

**Comment by MichelHollands** [comment]
That looks like a connection issue. Can you access loki-distributed-minio.test.svc:9000? If you use that same address for writes then perhaps it's a permissions problem for deletes.

**Comment by rt005744** [comment]
> 

  @MichelHollands  hi, How to set the automatic deletion of logs over 24 hours

**Comment by rt005744** [comment]
@MichelHollands @LinTechSo Automatic deletion of logs over 24 hours has not worked. Thank you very much for pasting the key configuration

**Comment by LinTechSo** [comment]
Yes. My connection is ok. and this problem sometimes occurred. I expected Loki to retry delete chunks at some other time 

> That looks like a connection issue. Can you access loki-distributed-minio.test.svc:9000? If you use that same address for writes then perhaps it's a permissions problem for deletes.

Yes. My connection is Ok. and this problem sometimes occurred. I expected Loki to use a retry retention cycle. but in my case, Loki didn't and just flooding `msg="error deleting chunk"`after two days.

**Comment by LinTechSo** [comment]
> @MichelHollands @LinTechSo Automatic deletion of logs over 24 hours has not worked. Thank you very much for pasting the key configuration

Hi. I think you should make sure that you are using compactor retention instead of table manager retention.

**Comment by rt005744** [comment]
> > @MichelHollands @LinTechSo Automatic deletion of logs over 24 hours has not worked. Thank you very much for pasting the key configuration
> 
> Hi. I think you should make sure that you are using compactor retention instead of table manager retention.

this is my config

compactor:
      working_directory: /var/loki/retention
      shared_store: s3
      compaction_interval: 10m
      retention_enabled: true
      retention_delete_delay: 2h
      retention_delete_worker_count: 150
........
table_manager:
      retention_deletes_enabled: true
      retention_period: 24h

**Comment by LinTechSo** [comment]
> > > @MichelHollands @LinTechSo Automatic deletion of logs over 24 hours has not worked. Thank you very much for pasting the key configuration
> > 
> > 
> > Hi. I think you should make sure that you are using compactor retention instead of table manager retention.
> 
> this is my config
> 
> compactor: working_directory: /var/loki/retention shared_store: s3 compaction_interval: 10m retention_enabled: true retention_delete_delay: 2h retention_delete_worker_count: 150 ........ table_manager: retention_deletes_enabled: true retention_period: 24h

Use this configuration instead

```
compactor:
working_directory: /var/loki/retention
shared_store: s3
compaction_interval: 10m
retention_enabled: true
retention_delete_delay: 2h
retention_delete_worker_count: 150
........
limits_config:
   retention_period: 24h
.......
table_manager: 
  retention_deletes_enabled: false

```

can't be both

**Comment by rt005744** [comment]
> > > > @MichelHollands @LinTechSo Automatic deletion of logs over 24 hours has not worked. Thank you very much for pasting the key configuration
> > > 
> > > 
> > > Hi. I think you should make sure that you are using compactor retention instead of table manager retention.
> > 
> > 
> > this is my config
> > compactor: working_directory: /var/loki/retention shared_store: s3 compaction_interval: 10m retention_enabled: true retention_delete_delay: 2h retention_delete_worker_count: 150 ........ table_manager: retention_deletes_enabled: true retention_period: 24h
> 
> Use this configuration instead
> 
> ```
> compactor:
>     working_directory: /var/loki/retention
>     shared_store: s3
>     compaction_interval: 10m
>     retention_enabled: true
>     retention_delete_delay: 2h
>     retention_delete_worker_count: 150
> ........
> limits_config:
>     retention_period: 24h
> .......
> table_manager: 
>    retention_deletes_enabled: false
> ```
> 
> can't be both
Thank you for your reply.

**Comment by LinTechSo** [comment]
OK. I close this issue because:
1 . Access to deletion API solved in my Loki version 2.6.1  
2. Retention issue occurred sometimes to me and now it works. if it doesn't become critical again I think we can discuss on that in another GitHub issue or slack channel.

**Comment by LinTechSo** [comment]
@MichelHollands I would appreciate it if you help me figure this `msg=" error deleting chunk"` out. and what can I do to decrease the possibility of this error?
Thanks a lot.
